hammr/utils/deployment_utils.py

```diff
@@ -72,134 +76,144 @@ def validate_deployment(file):
     except IOError as e:
         printer.out("unknown error deployment json file", printer.ERROR)
 
-def build_deployment_amazon(data):
-    data = validate_deployment(data)
+
+def check_and_get_attributes_from_file(file, expectedAttributes):
+    attributes = validate_deployment(file)
+    for attribute in expectedAttributes:
+        if not attribute in attributes:
+            raise TypeError("There is no attribute [" + attribute + "] for the provisioner")
+
+    return attributes
+
+
+def build_deployment_amazon(attributes):
     deployment = Deployment()
-    myinstance = InstanceAmazon()
-    if not "name" in data:
-        printer.out("There is no attribute [name] for the provisioner", printer.ERROR)
-        return None
-    deployment.name = data["name"]
-    if not "cores" in data:
-        myinstance.cores = "1"
+    myInstance = InstanceAmazon()
+
+    deployment.name = attributes["name"]
+    if not "cores" in attributes:
+        myInstance.cores = "1"
     else:
-        myinstance.cores = data["cores"]
-    if not "memory" in data:
-        myinstance.memory = "1024"
+        myInstance.cores = attributes["cores"]
+    if not "memory" in attributes:
+        myInstance.memory = "1024"
     else:
-        myinstance.memory = data["memory"]
+        myInstance.memory = attributes["memory"]
+
     deployment.instances = pyxb.BIND()
     deployment.instances._ExpandedName = pyxb.namespace.ExpandedName(Namespace, 'Instances')
-    deployment.instances.append(myinstance)
+    deployment.instances.append(myInstance)
     return deployment
 
-def build_deployment_azure(data):
-    data = validate_deployment(data)
+
+def build_deployment_azure(attributes):
     deployment = Deployment()
-    myinstance = InstanceAzureResourceManager()
-    if not "name" in data:
-        printer.out("There is no attribute [name] for the provisioner", printer.ERROR)
-        return None
-    deployment.name = data["name"]
-    if "userName" in data:
-        myinstance.userName = data["userName"]
-    else:
-        printer.out("There is no attribute [userName] for the provisioner", printer.ERROR)
-        return None
-    if "userSshKey" in data:
-        myinstance.userSshKey = data["userSshKey"]
-    elif "userSshKeyFile" in data:
-        try:
-            myinstance.userSshKey = open(data["userSshKeyFile"], "r").read()
-        except IOError as e:
-                printer.out("File error: "+str(e), printer.ERROR)
-                return
+    myInstance = InstanceAzureResourceManager()
+
+    deployment.name = attributes["name"]
```

Comment:
Behavior has changed in case of missing attributes. See integration tests which are failing